src/main/scala/scalaz/parsers/transform.scala

```diff
@@ -5,7 +5,18 @@ import scalaz.parsers.implicits.monadKleisliCategory
 import scalaz.tc.ProfunctorClass.DeriveDimap
 import scalaz.tc._
 
-trait TransformClass[=>:[_, _]] extends StrongClass[=>:] with CategoryClass[=>:] {
+trait CategoryClassOps[=>:[_, _]] extends { self: CategoryClass[=>:] =>
```

Comment:
`extends { self: CategoryClass[=>:]` + `with CategoryClassOps[=>:]` is a circular reference 🔃.
What is the advantage over usual syntax extension encoding?